packages/cli/src/number-parse.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Parses the input and returns it as a number if it's valid, otherwise returns undefined.
+ *
+ * @param {string | number} input
+ * @returns {number | undefined}
+ */
+export const parseNumber = input => {
+  if (typeof input === 'number' && !Number.isNaN(input)) {
+    return input;
+  }
+
+  if (typeof input === 'string' && /^-?\d+(\.\d+)?$/.test(input.trim())) {
+    return Number(input);
+  }
+
+  return undefined;
```

Comment:
Would it not be better to throw an error here?

Reply:
I also considered throwing an error, but given that we handle input validation later on, I opted for a more lenient approach to just extract numbers from strings. I can change it if you think it's better to throw instead.

Reply:
Hmm, I appreciate that thinking, but for the sake of consistency with [this file](https://github.com/endojs/endo/blob/kumavis-gems/packages/cli/src/pet-name.js) and on the principle that the daemon's unit tests should cover our input validation, let's just throw an error early.